arangod/Aql/ExecutionNode.h

```diff
@@ -935,56 +935,101 @@ class NoResultsNode : public ExecutionNode {
   CostEstimate estimateCost() const override final;
 };
 
-class MaterializeNode  : public ExecutionNode {
-  friend class ExecutionNode;
-  friend class ExecutionBlock;
-
- public:
-  MaterializeNode(ExecutionPlan* plan, size_t id, aql::Variable const& inColPtr,
-                   aql::Variable const& inDocId, aql::Variable const& outVariable);
+class MaterializeNode : public ExecutionNode {
+ protected:
+  MaterializeNode(ExecutionPlan* plan, size_t id, aql::Variable const& inDocId, aql::Variable const& outVariable);
 
   MaterializeNode(ExecutionPlan* plan, arangodb::velocypack::Slice const& base);
 
+ public:
+  static MaterializeNode* create(ExecutionPlan* plan, arangodb::velocypack::Slice const& base);
+
   /// @brief return the type of the node
-  NodeType getType() const override final { return ExecutionNode::MATERIALIZE; }
+  NodeType getType() const final { return ExecutionNode::MATERIALIZE; }
 
   /// @brief export to VelocyPack
   void toVelocyPackHelper(arangodb::velocypack::Builder& nodes, unsigned flags,
-                          std::unordered_set<ExecutionNode const*>& seen) const override final;
+                          std::unordered_set<ExecutionNode const*>& seen) const override;
 
   /// @brief creates corresponding ExecutionBlock
   std::unique_ptr<ExecutionBlock> createBlock(
       ExecutionEngine& engine,
-      std::unordered_map<ExecutionNode*, ExecutionBlock*> const&) const override;
+      std::unordered_map<ExecutionNode*, ExecutionBlock*> const&) const override = 0;
 
   /// @brief clone ExecutionNode recursively
   ExecutionNode* clone(ExecutionPlan* plan, bool withDependencies,
-                       bool withProperties) const override final;
+                       bool withProperties) const override = 0;
 
-  CostEstimate estimateCost() const override final;
+  CostEstimate estimateCost() const final;
```

Comment:
why isn't this `override` anymore?
```suggestion
  CostEstimate estimateCost() const override final;
```